services/jira/jira-issue.service.js

```diff
@@ -1,100 +1,92 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const serverSecrets = require('../../lib/server-secrets')
+const Joi = require('joi')
+const BaseJiraService = require('./jira-base')
 
-module.exports = class JiraIssue extends LegacyService {
-  static get category() {
-    return 'issue-tracking'
+const schema = Joi.object({
+  fields: Joi.object({
+    status: Joi.object({
+      name: Joi.string().required(),
+      statusCategory: Joi.object({
+        colorName: Joi.string().required(),
+      }),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class JiraIssue extends BaseJiraService {
+  static render({ issue, statusName, color }) {
+    return {
+      label: issue,
+      message: statusName,
+      color,
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { color: 'lightgrey', label: 'jira' }
   }
 
   static get route() {
     return {
       base: 'jira/issue',
+      pattern: ':protocol(https?)/:host/:path?/:issueKey',
```

Comment:
- The regex in this pattern confused me for a moment. `:protocol(http|https)` seems clearer.
- I think you need `:path*` to allow multiple path segments.

Reply:
👍 on protocol

`path` is where I kept hitting a a mental wall with the pattern. I was trying to use nested non-capture groups but pattern was having none of it 😆 Let me see if I can clean that up a bit

Reply:
I'm fine with the change you made, though I think `hostAndPath` would better describe the field for the user (this is with #2496 in mind).

Reply:
yeah I like that better too (and that's what was there before so +1 for consistency!)

Reply:
i'm probably being overly pedantic here, but just noticed the example pattern before was actually `hostAndPath+` does the `+` suffix have any significance in the example patterns/is it worth including?